src/test/util/logging.cpp

```diff
@@ -21,11 +22,12 @@ DebugLogHelper::DebugLogHelper(std::string message, MatchFn match)
     noui_test_redirect();
 }
 
-void DebugLogHelper::check_found()
+DebugLogHelper::~DebugLogHelper()
 {
     noui_reconnect();
     LogInstance().DeleteCallback(m_print_connection);
     if (!m_found && m_match(nullptr)) {
-        throw std::runtime_error(strprintf("'%s' not found in debug log\n", m_message));
+        tfm::format(std::cerr, "Fatal error: expected message not found in the debug log: '%s'\n", m_message);
```

Comment:
nit:
"Fatal error: expected message '%s' not found in the logs\n"
otherwise it reads as you were printing the debug log.

Reply:
Will do if I retouch. Change applied locally.